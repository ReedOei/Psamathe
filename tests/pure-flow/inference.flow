;

[] --> var m : map nat => nat

{ k |-> 0, v |-> 0 } --> var x : { k : nat, v : nat }

