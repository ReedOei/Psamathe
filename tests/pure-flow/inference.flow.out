<generatedTop>
  <common>
    <k>
      .
    </k>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
    <functions>
      .Map
    </functions>
  </common>
  <lookup>
    m |-> selected ( loc ( 0 ) , loc ( 0 ) )
    x |-> selected ( loc ( 1 ) , loc ( 1 ) )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( table ( key , .Vars ) one record ( key , .Vars ) { key : any nat , value : any nat , .VarDefs } , .List )
    loc ( 1 ) |-> resource ( record ( .Vars ) { k : any nat , v : any nat , .VarDefs } , ListItem ( k |-> loc ( 4 )
    v |-> loc ( 6 ) ) )
    loc ( 2 ) |-> resource ( record ( .Vars ) { k : any nat , v : any nat , .VarDefs } , .List )
    loc ( 3 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 4 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 5 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 6 ) |-> resource ( nat , ListItem ( 0 ) )
  </storage>
  <catchStack>
    .List
  </catchStack>
  <createdVals>
    .Map
  </createdVals>
</generatedTop>
